syntax = "proto3";

package notcoolj.transaction.v1;

option go_package = "github.com/cryptow";

service FetcherService {
  rpc FetchTrades(FetchTradesRequest) returns (FetchTradesResponse) {}

  rpc StopFetchTrades(Empty) returns (StopFetchTradesResponse) {}
}

message DataSource {
  string source = 1;
}

message FetchTradesRequest {
  repeated DataSource sources = 1;
}

message Task {
  int32 id = 1;
}

message FetchTradesResponse {
  repeated Task tasks = 1;
}

message StopFetchTradesResponse {
  bool success = 1;
}

message Empty {

}